### PR TITLE
fix: table-editor-negative-bigint-filter-precision

### LINKED ISSUE #47470: Bug: Table Editor filter on bigint columns returns wrong results for large negative values
## Describe the bug

In the Table Editor, filtering a `bigint` (`int8`) column by a large **negative** value
silently returns the wrong result set — the matching row does not appear. The equivalent
large **positive** value works correctly, so the behavior is asymmetric.

The root cause is in `formatFilterValue` (`apps/studio/data/table-rows/utils.ts`). It keeps
out-of-range bigint filter values as strings to avoid precision loss, but only checks the
upper bound of the JS safe-integer range:

```ts
const numberValue = Number(filter.value)
// Supports BigInt filter values
if (Number.isNaN(numberValue) || numberValue > Number.MAX_SAFE_INTEGER) return filter.value
else return Number(filter.value)
```

`numberValue > Number.MAX_SAFE_INTEGER` is always `false` for negative numbers, so large
negative bigints fall through and are rounded by `Number()`:

```
Number('-9223372036854775808') === -9223372036854776000   // rounded
```

The rounded value is then used to build the SQL filter, so it no longer matches the row the
user filtered for. The same helper also feeds the row count and "delete all matching"
queries.

## To Reproduce

1. Create a table with a `bigint` column, e.g. `id`.
2. Insert a row with a large negative id, e.g. `-9223372036854775808` (the int8 minimum).
3. In the Table Editor, add the filter `id = -9223372036854775808`.
4. Observe that the row is not returned.
5. For contrast, filter by the positive counterpart `9223372036854775807` — it works.

## Expected behavior

Large negative bigints should be preserved exactly (like large positive bigints), so the
literal sent to Postgres matches what the user typed and the correct rows are returned.

## System information

- Affected surface: Studio → Table Editor (also `table-rows-count-query` and
  `table-row-delete-all-mutation`, which share `formatFilterValue`)
- File: `apps/studio/data/table-rows/utils.ts`

## Additional context

Severity: medium — data-correctness bug affecting `bigint` columns for values whose
magnitude exceeds `Number.MAX_SAFE_INTEGER` (`9007199254740991`). No security impact. A
one-line fix (guarding the range by magnitude) plus unit tests resolves it.

### PR DESCRIPTION
**Pull request by Abuhaithem**
Fixes: #47470

## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

Bug fix.

## What is the current behavior?

In the Table Editor, filtering a `bigint` (`int8`) column by a large **negative** value
returns the wrong results (the matching row does not appear), while the equivalent large
**positive** value works correctly.

`formatFilterValue` (`apps/studio/data/table-rows/utils.ts`) keeps out-of-range bigint
filter values as strings so they reach Postgres without precision loss, but it only guards
the upper end of the JS safe-integer range:

```ts
const numberValue = Number(filter.value)
// Supports BigInt filter values
if (Number.isNaN(numberValue) || numberValue > Number.MAX_SAFE_INTEGER) return filter.value
else return Number(filter.value)
```

`numberValue > Number.MAX_SAFE_INTEGER` is always `false` for negative numbers, so large
negative bigints (e.g. the int8 minimum `-9223372036854775808`) fall through and get rounded
by `Number()` (`Number('-9223372036854775808')` → `-9223372036854776000`). The rounded value
is then sent to SQL, so the filter no longer matches the intended row. The same helper feeds
the row count and "delete all matching" queries.

Steps to reproduce:

1. Create a table with a `bigint` column `id`.
2. Insert a row with `id = -9223372036854775808`.
3. In the Table Editor, filter `id = -9223372036854775808`.
4. The row is not returned. Filtering by `9223372036854775807` works as expected.

## What is the new behavior?

Large negative bigints are now preserved as strings just like large positive ones, so the
literal sent to Postgres matches what the user typed and the filter returns the correct rows.

The fix guards the safe-integer range by magnitude:

```ts
if (Number.isNaN(numberValue) || Math.abs(numberValue) > Number.MAX_SAFE_INTEGER)
  return filter.value
else return numberValue
```

In-range values and large positive bigints are unaffected.

## Additional context

- Added unit tests in `apps/studio/data/table-rows/utils.test.ts` covering non-numerical
  passthrough, in-range coercion (positive and negative), `NaN` passthrough, large positive
  bigints (existing behavior), large negative bigints (regression), and the exact
  safe-integer bounds.
- The negative-bigint test fails on `master` and passes with this change.

Verify locally:

```bash
pnpm --filter studio exec vitest run data/table-rows/utils.test.ts
```

No API, schema, or infrastructure changes.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved filter value formatting to keep the original input when numeric conversion would be unsafe (invalid numbers or values outside safe-integer bounds), including large negative inputs.

* **Tests**
  * Added automated coverage for filter value formatting across non-numeric values, valid numeric coercion, invalid numeric strings, and bigint-like edge cases (including a large negative regression case).

### CODE REVIEW

File: apps/studio/data/table-rows/utils.ts [hunk 1]
```diff
@@ -18,9 +18,15 @@ export function formatFilterValue(
...
     const numberValue = Number(filter.value)
-    // Supports BigInt filter values
-    if (Number.isNaN(numberValue) || numberValue > Number.MAX_SAFE_INTEGER) return filter.value
-    else return Number(filter.value)
+    // Keep the original string for values that can't be represented exactly as
+    // a JS number (e.g. large bigints), otherwise Number() silently loses
+    // precision and we'd filter on the wrong value. This must guard both the
+    // positive and negative ends of the safe-integer range — checking only the
+    // upper bound let large negative bigints (e.g. the int8 minimum
+    // -9223372036854775808) slip through and get rounded.
```
Comment (awaseem) [inline apps/studio/data/table-rows/utils.ts]: Can we remove this comment, its super verbose

**Review by awaseem — CHANGES_REQUESTED** [review]

**Review by awaseem — APPROVED** [review]
